src/main/kotlin/pt/up/fe/ni/website/backend/config/upload/UploadConfig.kt

```diff
@@ -0,0 +1,33 @@
+package pt.up.fe.ni.website.backend.config.upload
+
+import com.cloudinary.Cloudinary
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.util.ResourceUtils
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer
+import pt.up.fe.ni.website.backend.util.CloudinaryFileUploader
+import pt.up.fe.ni.website.backend.util.FileUploader
+import pt.up.fe.ni.website.backend.util.StaticFileUploader
+
+@Configuration
+class UploadConfig(
+    private val uploadConfigProperties: UploadConfigProperties,
+
+) : WebMvcConfigurer {
+    @Bean
+    fun fileUploader(): FileUploader {
+        return when (uploadConfigProperties.uploadType) {
+            "cloudinary" -> CloudinaryFileUploader(
+                uploadConfigProperties.cloudinaryBasePath ?: "/",
+                Cloudinary(
+                    uploadConfigProperties.cloudinaryUrl
+                        ?: kotlin.run { throw Error("Cloudinary URL not provided") }
+                )
```

Comment:
Maybe adding a variable here is more readable and avoids `kotlin.run`

Reply:
Other constructor params are simpler so I don't mind

Reply:
I don't think i understand. Can you clarify?

Reply:
I meant something like this (form chatgpt, not tested)

````kotlin
fun fileUploader(): FileUploader {
    return if (uploadConfigProperties.provider == "cloudinary") {
        val cloudinaryBaseUrl = uploadConfigProperties.cloudinaryBasePath ?: "/"
        val cloudinaryUrl = uploadConfigProperties.cloudinaryUrl
            ?: throw Error("Cloudinary URL not provided")
        CloudinaryFileUploader(cloudinaryBaseUrl, Cloudinary(cloudinaryUrl))
    } else {
        val staticPath = uploadConfigProperties.staticPath?.let { ResourceUtils.getFile(it).absolutePath } ?: ""
        val staticServe = uploadConfigProperties.staticServe ?: "localhost:8080"
        StaticFileUploader(staticPath, staticServe)
    }
}
````